core/consensus/slots/src/lib.rs

```diff
@@ -132,43 +143,44 @@ pub fn start_slot_worker<B, C, W, SO, SC, OnExit>(
 	SO: SyncOracle + Send + Clone,
 	SC: SlotCompatible,
 	OnExit: Future<Item=(), Error=()>,
+	T: SlotData + Clone,
 {
-	worker.on_start(slot_duration.0)?;
+	worker.on_start(slot_duration.slot_duration())?;
 
 	let make_authorship = move || {
 		let client = client.clone();
 		let worker = worker.clone();
 		let sync_oracle = sync_oracle.clone();
-		let SlotDuration(slot_duration) = slot_duration;
+		let SlotDuration(slot_duration) = slot_duration.clone();
```

Comment:
`slot_duration` doesn't need cloning, does it? it's shadowed immediately after and the original value is not used anymore

Reply:
Actually, it does:

    error[E0507]: cannot move out of captured variable in an `FnMut` closure
       --> core/consensus/slots/src/lib.rs:189:56
        |
    150 |     let make_authorship = move || {
        |         --------------- captured outer variable
    ...
    189 |         let authorship_task = ::std::panic::AssertUnwindSafe(make_authorship());
        |                                                              ^^^^^^^^^^^^^^^ cannot move out of captured variable in an `FnMut` closure